airflow/www/templates/airflow/dag.html

```diff
@@ -315,6 +315,19 @@ <h4 class="modal-title" id="dagModalLabel">
     var task_id = '';
     var exection_date = '';
     var subdag_id = '';
+
+    var buttons = {
+      subDag: {elem: document.querySelector('#btn_subdag'), url: '{{ url_for( 'Airflow.' + dag.get_default_view() ) }}'},
```

Comment:
I have concerns about mixing JS and Python/Jinja code. Javascript is a dynamic language, so if we generate Javascript code using a different language, we do meta-programming. This drastically increases the complexity and makes testing difficult. Second, the Javascript code should be separated from the HTML code in the long-term to allow independent testing. Now it is not possible, but it is possible that one day it will succeed. However, if we have Python/Jinja code in Javascript it will be drastically difficult. HTML code is a language to describe the structure of a document and is better suited for storing this type of information.

I prepared an example of how in my opinion it can be separated:
https://codepen.io/anon/pen/LqKPKZ?editors=1010
What do you think about my idea? 

I am very happy that someone wants to improve the HTML and JS code. This is on my dream list, but the lack of automatic tests scares me away.

Reply:
The current code unfortunately does this already, so I don't think this PR makes things worse. But I agree that it would be better to separate jinja and js. I'll try refactoring this and see how it goes.

Reply:
@mik-laj In your example you set a data-href attribute to a URL, and then have JS that set's the href based on that - why not set href directly?